src/index.ts

```diff
@@ -370,7 +370,8 @@ export class Components implements OnInit, OnStart, OnTick, OnPhysics, OnRender
 		if (skipInstanceCheck !== true) {
 			const instanceGuard = this.getInstanceGuard(component);
 			if (instanceGuard !== undefined) {
-                if (instanceGuard(instance) === false) {
+                const result = instanceGuard(instance);
+                if (result === false || result === undefined) {
```

Comment:
Reference: https://developer.roblox.com/en-us/api-reference/lua-docs/Lua-Globals
`Throws an error if the provided value is false or nil. If the assertion passes, it returns all values passed to it.`